core/src/main/java/hudson/model/FreeStyleProject.java

```diff
@@ -47,6 +48,7 @@ public FreeStyleProject(Jenkins parent, String name) {
         super(parent, name);
     }
 
+    @DataBoundConstructor
```

Comment:
This is not right. It is not intended to be instantiated like a common `Describable`. Calls to `TopLevelItemDescriptor.newInstance(ItemGroup, String)` (which is the official API here, _not_ direct constructor calls) must be made explicitly by `job-dsl`. The situation is similar to a `GlobalConfiguration`, for example, where you may define `@DataBoundSetter`s for modifying properties of an existing object, but a special code path is used to actually get to the initial default instance.

Reply:
Or for the reverse case, of introspecting a `TopLevelItem` and enumerating its properties as a `Map<String, Object>` (_not_ an `UninstantiatedDescribable`), we would need a new API in `structs`.